emulator.c

```diff
@@ -380,7 +427,23 @@ execute_instruction(i8080 *cpu, uint8_t opcode)
         address += (cpu_read_mem(cpu, cpu->sp + 1) << 8); // NOLINT
         cpu->sp += 2;
         cpu->pc = address;
-        return 0;
+        // NOLINTNEXTLINE
+        return 10; // return num cycles
+      }
+    case 0xca: // NOLINT
+      {        // JZ
+        if (is_zero_flag_set(cpu))
+          {
+            uint16_t address = cpu_read_mem(cpu, cpu->pc + 2); // NOLINT
+            address = address << 8;                            // NOLINT
+            address += cpu_read_mem(cpu, cpu->pc + 1);
+            cpu->pc = address;
+            // NOLINTNEXTLINE
+            return 10; // return num cycles
+          }
+        cpu->pc += 2;    // NOLINT
+        num_cycles = 10; // NOLINT
```

Comment:
Interesting how this one has the same cycle count regardless of jump or not compared to 0xc8 which is different. As far as I can tell this is correct.

Reply:
Yeah our sources definitely seem to indicate when this varies...